Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -41,31 +46,27 @@ final class SitterProfileViewModel: ObservableObject {
     }
 
     /// Requests model layer to upload and save modified data.
-    func save() async {
-        await MainActor.run {
-            isSavingData = true
-        }
-
-        var updatedSitter = currentSitter
+    @MainActor func save() async {
+        isSavingData = true
 
-        updatedSitter.name = name
-        updatedSitter.surname = surname
-        updatedSitter.phone = phone
-        updatedSitter.bio = bio
-        updatedSitter.pricePerHour = Double(pricePerHour) ?? currentSitter.pricePerHour
+        var newSitter = Sitter()
+        newSitter.name = name
+        newSitter.surname = surname
+        newSitter.phone = phone
+        newSitter.bio = bio
+        newSitter.pricePerHour = Double(pricePerHour) ?? 0
 
         do {
-            try await upload(updatedSitter)
-
-            currentSitter = updatedSitter
-
-            await saveLocally(currentSitter)
+            try await upload(newSitter)
+            currentSitter = newSitter
+            if let currentSitter {
+                try await saveLocally(currentSitter)
+            }
```

Comment:
Why do you prefer saving the current sitter instead of a new one? Is there any case that they can be different?

Reply:
You're right it doesn't matter. Deleted with e9d20b2